pyomo/solvers/plugins/solvers/cplex_direct.py

```diff
@@ -191,6 +192,7 @@ def _process_stream(arg):
 
             t1 = time.time()
             self._wallclock_time = t1 - t0
+            self._deterministic_time = self._solver_model.get_dettime()
```

Comment:
@begunefeoglu I believe `get_dettime()` just returns a [timestamp](https://www.ibm.com/support/knowledgecenter/SSSA5P_12.7.0/ilog.odms.cplex.help/refjavacplex/html/ilog/cplex/IloCplex.html#getDetTime()), not the actual result in ticks.

> To measure deterministic time spent between a starting point and ending point of an operation, take the time stamp at the starting point; take the time stamp at the ending point; subtract the starting time stamp from the ending time stamp to get elapsed time in deterministic ticks.
>
> The absolute value of the deterministic time stamp is not meaningful.



Note this can also be copied to the PersistentSolver

Reply:
Good catch, thanks!